subsys/disk/disk_access_sdhc.h

```diff
@@ -0,0 +1,751 @@
+/* disc_access_sdhc.h - header SDHC*/
+
+/*
+ * Copyright (c) 2019 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DISK_ACCESS_SDHC_H_
+#define ZEPHYR_INCLUDE_DISK_ACCESS_SDHC_H_
+
+#include <spi.h>
+
+#define SDMMC_CLOCK_400KHZ (400000U)
+#define SD_CLOCK_25MHZ (25000000U)
+#define SD_CLOCK_50MHZ (50000000U)
+#define SD_CLOCK_100MHZ (100000000U)
+#define SD_CLOCK_208MHZ (208000000U)
+#define MMC_CLOCK_26MHZ (26000000U)
+#define MMC_CLOCK_52MHZ (52000000U)
+#define MMC_CLOCK_DDR52 (52000000U)
+#define MMC_CLOCK_HS200 (200000000U)
+#define MMC_CLOCK_HS400 (400000000U)
+
+/* Command IDs */
+#define SDHC_GO_IDLE_STATE 0
+#define SDHC_ALL_SEND_CID 2
+#define SDHC_SEND_RELATIVE_ADDR 3
+#define SDHC_SWITCH 6
+#define SDHC_SELECT_CARD 7
+#define SDHC_SEND_IF_COND 8
+#define SDHC_SEND_CSD 9
+#define SDHC_SEND_CID 10
+#define SDHC_VOL_SWITCH 11
+#define SDHC_STOP_TRANSMISSION 12
+#define SDHC_SEND_STATUS 13
+#define SDHC_GO_INACTIVE_STATE 15
+#define SDHC_SET_BLOCK_SIZE 16
+#define SDHC_READ_SINGLE_BLOCK 17
+#define SDHC_READ_MULTIPLE_BLOCK 18
+#define SDHC_SEND_TUNING_BLOCK 19
+#define SDHC_SET_BLOCK_COUNT 23
+#define SDHC_WRITE_BLOCK 24
+#define SDHC_WRITE_MULTIPLE_BLOCK 25
+#define SDHC_ERASE_BLOCK_START 32
+#define SDHC_ERASE_BLOCK_END 33
+#define SDHC_ERASE_BLOCK_OPERATION 38
+#define SDHC_APP_CMD 55
+#define SDHC_READ_OCR 58
+#define SDHC_CRC_ON_OFF 59
+
+enum sdhc_app_ext_cmd {
+	SDHC_APP_SET_BUS_WIDTH = 6,
+	SDHC_APP_SEND_STATUS = 13,
+	SDHC_APP_SEND_NUM_WRITTEN_BLK = 22,
+	SDHC_APP_SET_WRITE_BLK_ERASE_CNT = 23,
+	SDHC_APP_SEND_OP_COND = 41,
+	SDHC_APP_CLEAR_CARD_DETECT = 42,
+	SDHC_APP_SEND_SCR = 51
+};
+
+#define SDHC_SEND_OP_COND SDHC_APP_SEND_OP_COND
+
+/* R1 response status */
+#define SDHC_R1_IDLE 0x01
+#define SDHC_R1_ERASE_RESET 0x02
+#define SDHC_R1_ILLEGAL_COMMAND 0x04
+#define SDHC_R1_COM_CRC 0x08
+#define SDHC_R1_ERASE_SEQ 0x10
+#define SDHC_R1_ADDRESS 0x20
+#define SDHC_R1_PARAMETER 0x40
+
+#define SDHC_CMD_SIZE 6
+#define SDHC_CMD_BODY_SIZE (SDHC_CMD_SIZE - 1)
+#define SDHC_CRC16_SIZE 2
+
+/* Command flags */
+#define SDHC_START 0x80
+#define SDHC_TX 0x40
+
+/* Fields in various card registers */
+#define SDHC_HCS (1 << 30)
+#define SDHC_CCS (1 << 30)
+#define SDHC_VHS_MASK (0x0F << 8)
+#define SDHC_VHS_3V3 (1 << 8)
+#define SDHC_CHECK 0xAA
+#define SDHC_CSD_SIZE 16
+#define SDHC_CSD_V2 1
+
+/* Data block tokens */
+#define SDHC_TOKEN_SINGLE 0xFE
+#define SDHC_TOKEN_MULTI_WRITE 0xFC
+#define SDHC_TOKEN_STOP_TRAN 0xFD
+
+/* Data block responses */
+#define SDHC_RESPONSE_ACCEPTED 0x05
+#define SDHC_RESPONSE_CRC_ERR 0x0B
+#define SDHC_RESPONSE_WRITE_ERR 0x0E
+
+/* Clock speed used during initialisation */
+#define SDHC_INITIAL_SPEED 400000
+/* Clock speed used after initialisation */
+#define SDHC_SPEED 4000000
```

Comment:
SDHC_SPEED should be moved to kconfig or dts.

Or have a bus speed defined and do the math from there to determine speed? What do you think?

Reply:
SDHC_INITIAL_SPEED and SDHC_SPEED are only used for spi sdhc to configure spi speed.
I will rename it to avoid confusing and move it from generic header to disk_access_sdhc_spi.c.

Reply:
Done